packages/rocketchat-lib/server/lib/notifyUsersOnMessage.js

```diff
@@ -104,9 +104,24 @@ RocketChat.callbacks.add('afterSaveMessage', function(message, room) {
 	// Update all the room activity tracker fields
 	RocketChat.models.Rooms.incMsgCountAndSetLastMessageById(message.rid, 1, message.ts, RocketChat.settings.get('Store_Last_Message') && message);
 
-	// Update all other subscriptions to alert their owners but witout incrementing
+	// Update all other subscriptions to alert their owners but without incrementing
 	// the unread counter, as it is only for mentions and direct messages
 	RocketChat.models.Subscriptions.setAlertForRoomIdExcludingUserId(message.rid, message.u._id);
+	const subscriptions = RocketChat.models.Subscriptions.findByRoomIdForAlert(message.rid);
+	if (subscriptions) {
+		let userSubscribed = {};
+		let showHiddenRoom;
+		subscriptions.forEach(function(subscription) {
+			if (subscription.u._id !== message.u._id) {
+				userSubscribed = RocketChat.models.Users.findById(message.u._id);
```

Comment:
Is that correct? Get the message user and not the subscription user?

Reply:
`findById` is a find, so it returns a cursor, not one record, did you tested this?

Reply:
Fixed!